### PR TITLE
[SBOM] fix:remove cache cleaner in collector and duplicated telemetry metrics. Change default options.

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
Removes the goroutine that cleans the cache because a scan can not run in parallel with a cache clean.
Some metrics were duplicated by this [PR](https://github.com/DataDog/datadog-agent/pull/16612). This PR moves all sbom metrics from `workloadmeta/telemetry` to `sbom/telemetry`. 
This PR also updates the default options of the SBOM cache.

### Motivation
Avoid confusion with duplicated metrics. Avoid cache entries deletion during a scan. Use a cache big enough to store all the blobs.

### Additional Notes
It's currently impossible to test the changes as SBOM collection does not work on main.

### Possible Drawbacks / Trade-offs
N/A

### Describe how to test/QA your changes
Run the agent with SBOM collection enabled and `DD_TELEMETRY_ENABLED="true"`.
Retrieve the sbom metrics with `curl -s telemetry localhost:6000/telemetry | grep -i sbom`.
SBOM metrics should not be prefixed by `workloadmeta`.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/workloadmeta/telemetry/telemetry.go
Comment (L3n41c) [inline pkg/workloadmeta/telemetry/telemetry.go:1]: `sbom_generation_duration` is also already defined in `pkg/sbom/telemetry/telemetry.go`, so should we remove it from this file as well ?

```go
	// SBOMGenerationDuration measures the time that it takes to generate SBOMs
	// in seconds.
	SBOMGenerationDuration = telemetry.NewHistogramWithOpts(
		subsystem,
		"sbom_generation_duration",
		[]string{},
		"SBOM generation duration (in seconds)",
		[]float64{10, 30, 60, 120, 180, 240, 300, 360, 420, 480, 540, 600},
		commonOpts,
	)
```

Comment (AliDatadog) [reply pkg/workloadmeta/telemetry/telemetry.go:1]: Good catch, it should be fixed now

File: pkg/sbom/telemetry/telemetry.go [hunk 1]
```diff
@@ -48,15 +50,60 @@ var (
...
+		"sbom_cached_keys",
+		[]string{},
+		"Number of cache keys stored in memory",
+		commonOpts,
+	)
+
+	// SBOMCachedObjectSize total size of cached objects in disk (in bytes) used for SBOM collection
+	SBOMCachedObjectSize = telemetry.NewGaugeWithOpts(
+		subsystem,
+		"sbom_cached_object_size",
```
[the remark this answers is not in the document]
Comment (AliDatadog) [reply pkg/sbom/telemetry/telemetry.go]: I also realized that `subsystem` corresponds to `sbom`. I don't think the metrics should be `sbom.sbom_*` so I renamed them as well.

**Comment by AliDatadog** [comment]
I changed the default cache options in the config because they were too limiting as well.

**Review by pgimalac — APPROVED** [review]
LGTM for config file.

**Review by lebauce — APPROVED** [review]

**Review by cswatt — APPROVED** [review]

File: pkg/config/config.go [hunk 2]
```diff
@@ -1090,6 +1090,7 @@ func InitConfig(config Config) {
 
 	// SBOM configuration
 	bindEnvAndSetLogsConfigKeys(config, "sbom.")
+
```
Comment (AliDatadog) [inline pkg/config/config.go:1093]:
```suggestion
```

File: pkg/config/config.go [hunk 3]
```diff
@@ -1662,12 +1662,12 @@ func setupFipsLogsConfig(config Config, configPrefix string, url string) {
 func setupSBOMConfig(config Config, cacheDir string) {
 	config.BindEnvAndSetDefault("sbom.enabled", false)
 	config.BindEnvAndSetDefault("sbom.analyzers", []string{"os"})
-	config.BindEnvAndSetDefault("sbom.cache_directory", filepath.Join(defaultRunPath, cacheDir))
+	config.BindEnvAndSetDefault("sbom.cache_directory", filepath.Join(defaultRunPath, cacheDir))
```
Comment (AliDatadog) [inline pkg/config/config.go:1665]:
```suggestion
	config.BindEnvAndSetDefault("sbom.cache_directory", filepath.Join(defaultRunPath, cacheDir))
```

**Review by L3n41c — APPROVED** [review]

**Pull request merged**